dev-client/src/model/sync/sync.ts

```diff
@@ -15,13 +15,13 @@
  * along with this program. If not, see https://www.gnu.org/licenses/.
  */
 
-export type ChangeRecords<T> = Record<string, ChangeRecord<T>>;
+export type ChangeRecords<T, E> = Record<string, ChangeRecord<T, E>>;
 
 export type ChangeTimestamp = number;
 
 export type ChangeRevisionId = number;
 
-export type ChangeRecord<T> = {
+export type ChangeRecord<T, E> = {
```

Comment:
I see your point. I think a compromise might be renaming T to D so it clearly implies "data" and "error"; originally it was just T before I added the error param and I didn't bother to update it. I think for several parameter types it can be helpful to be more clear, yeah.